db/pf-schema-X.Y.Z.sql

```diff
@@ -1528,6 +1530,127 @@ CREATE TABLE `pki_revoked_certs` (
+-- Table structure for table `bandwidth_accounting`
+--
+
+CREATE TABLE bandwidth_accounting (
+    tenant_id INT,
+    mac char(17) NOT NULL,
+    unique_session_id char(32),
+    time_bucket DATETIME NOT NULL,
+    in_bytes BIGINT UNSIGNED NOT NULL,
+    out_bytes BIGINT UNSIGNED NOT NULL,
+    total_bytes BIGINT UNSIGNED AS (in_bytes + out_bytes) PERSISTENT,
+    processed BOOLEAN NOT NULL DEFAULT 0,
+    PRIMARY KEY (tenant_id, mac, time_bucket, unique_session_id),
+    KEY bandwidth_aggregate_buckets (time_bucket, tenant_id, mac, unique_session_id, in_bytes, out_bytes )
+);
+
+--
+-- Table structure for table `bandwidth_accounting_history`
+--
+
+CREATE TABLE bandwidth_accounting_history (
+    tenant_id INT,
+    mac char(17) NOT NULL,
+    time_bucket DATETIME NOT NULL,
+    in_bytes BIGINT UNSIGNED NOT NULL,
+    out_bytes BIGINT UNSIGNED NOT NULL,
+    total_bytes BIGINT UNSIGNED AS (in_bytes + out_bytes) PERSISTENT,
+    PRIMARY KEY (tenant_id, mac, time_bucket),
+    KEY bandwidth_aggregate_buckets (time_bucket, tenant_id, mac, in_bytes, out_bytes )
+);
+
+DROP PROCEDURE IF EXISTS `bandwidth_aggregation`;
+DELIMITER /
+CREATE PROCEDURE `bandwidth_aggregation` (
+  IN `p_bucket_size` varchar(255),
+  IN `p_end_bucket` datetime,
+  IN `p_batch` int(11) unsigned
+)
+BEGIN
+
+    SET @end_bucket= p_end_bucket, @batch = p_batch;
+    SET @create_table_to_delete_stmt = CONCAT('CREATE TEMPORARY TABLE to_delete ENGINE=MEMORY, MAX_ROWS=', @batch, ' SELECT tenant_id, mac, time_bucket as new_time_bucket, time_bucket, unique_session_id, in_bytes, out_bytes FROM bandwidth_accounting LIMIT 0');
+    PREPARE create_table_to_delete FROM @create_table_to_delete_stmt;
+    EXECUTE create_table_to_delete;
+    DEALLOCATE PREPARE create_table_to_delete;
+    SET @date_trunc = CASE p_bucket_size 
+        WHEN 'daily' THEN 'DATE(time_bucket)' 
+        WHEN 'monthly' THEN 'DATE(time_bucket)' 
+        ELSE 'DATE_ADD(DATE(time_bucket), INTERVAL HOUR(time_bucket) HOUR)'
+    END;
+    PREPARE insert_into_to_delete FROM  'INSERT INTO to_delete SELECT tenant_id, mac, DATE_ADD(DATE(time_bucket), INTERVAL HOUR(time_bucket) HOUR) as new_time_bucket, time_bucket, unique_session_id, in_bytes, out_bytes FROM bandwidth_accounting WHERE time_bucket <= ? AND processed = 1 LIMIT ?';
+
+
+    START TRANSACTION;
+    EXECUTE insert_into_to_delete using @end_bucket, @batch;
+    SELECT COUNT(*) INTO @count FROM to_delete;
+    IF @count > 0 THEN
+
+        INSERT INTO bandwidth_accounting_history
+        (tenant_id, mac, time_bucket, in_bytes, out_bytes)
+         SELECT
+             tenant_id,
+             mac,
+             new_time_bucket,
+             sum(in_bytes) AS in_bytes,
+             sum(out_bytes) AS out_bytes
+            FROM to_delete
+            GROUP BY tenant_id, mac, new_time_bucket
+            ON DUPLICATE KEY UPDATE
+                in_bytes = in_bytes + VALUES(in_bytes),
+                out_bytes = out_bytes + VALUES(out_bytes)
+            ;
```

Comment:
`total_bytes` also needs to be aggregated to `bandwidth_accounting_history`.

Reply:
It is automatically since it is a generated field.
https://github.com/inverse-inc/packetfence/pull/5181/files/3c544a9f3fb69659dd5950f934ae9ec1cc501744#diff-9a3daa21c2b9052d5f85954a88f0592cR1560

https://mariadb.com/kb/en/generated-columns/